### ISSUE TITLE
Can't use match function in string 

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#40740]
**Godot version:**

4.0.dev.custom_build. 6de86946e

**Issue description:**

```
var q_String : String = ""

if randi() % 2 == 1:
	q_String.match("")
```
shows error
```
 res://ResourcesALL/String.gd:153 - Parse Error: Expected identifier after "." for attribute access.
 modules/gdscript/gdscript.cpp:608 - Method/function failed. Returning: ERR_PARSE_ERROR
```

**Comment by ThakeeNathees** [comment]
#36878 fixed after GDScript 2.0, keywords are reserved and the following code is invalid now
```gdscript
func if():
	print('if() called')

func _ready():
	self.if()
```
is this intended ?

**Comment by vnen** [comment]
> ```gdscript
> ```gdscript
> func if():
> 	print('if() called')
> 
> func _ready():
> 	self.if()
> ```
> 
> 
> is this intended ?
> ```

It is. This one wasn't even supposed to work before TBH. The exception was more for stuff like `sync`, and the built-in functions which were also treated as keywords. But now built-in functions are not special tokens and many keywords became annotation, so this isn't much needed anymore.

I would prefer that this remains reserved. But I guess we can make an exception for a few keywords (such as `match` in this example). I added a TODO for this: https://github.com/godotengine/godot/blob/a1ca21f46d41d5528bb43ba15f2f370d78a823c4/modules/gdscript/gdscript_tokenizer.h#L171
